docs/recipes/install/poc-markdown/common/parse_doc_md.py

```diff
@@ -0,0 +1,314 @@
+
+        return cmd
+
+    def _process_here_doc(self, match: re.Match, output: StringIO) -> None:
+        """Process HERE document."""
+        cmd = self._substitute_variables(match.group(1))
+        delimiter = match.group(2).strip()
+
+        output.write(f"{cmd}\n")
+
+        # Process remaining lines until delimiter
+        try:
+            while True:
+                line = next(self._line_iterator)
+                line = line.rstrip("\n\r").strip()
+
+                # Check if we hit code block end
+                if self._PATTERNS["code_block_end"].match(line):
+                    break
+
+                line = self._substitute_variables(line)
+                output.write(f"{line}\n")
+                if line.startswith(delimiter):
+                    break
+        except StopIteration:
+            pass
+
+    def _process_continuation(self, match: re.Match, output: StringIO) -> None:
+        """Process line continuation."""
+        cmd = self._substitute_variables(match.group(1))
+        output.write(f"{cmd} \\")
+
+        try:
+            while True:
+                line = next(self._line_iterator)
+                line = line.rstrip("\n\r").strip()
+
+                # Check for code block end
+                if self._PATTERNS["code_block_end"].match(line):
+                    output.write("\n")
+                    break
+
+                # Skip [sms]# prefix if present
+                if line.startswith("[sms]#"):
+                    line = line[6:].strip()
+
+                line = self._substitute_variables(line)
+
+                if line.endswith("\\"):
+                    output.write(f"\n\t{line}")
+                else:
+                    output.write(f"\n\t{line}\n")
+                    break
+        except StopIteration:
+            output.write("\n")
+
+    def _process_do_loop(self, match: re.Match, output: StringIO) -> None:
+        """Process do loop with proper indentation."""
+        cmd = self._substitute_variables(match.group(1))
+        output.write(f"{cmd}\n")
+
+        try:
+            while True:
+                line = next(self._line_iterator)
+                line = line.rstrip("\n\r").strip()
+
+                # Check for code block end
+                if self._PATTERNS["code_block_end"].match(line):
+                    break
+
+                # Skip [sms]# prefix if present
+                if line.startswith("[sms]#"):
+                    line = line[6:].strip()
+
+                line = self._substitute_variables(line)
+
+                # Add indentation for lines inside the do loop
+                if not self._PATTERNS["done_line"].search(line):
+                    output.write(f"   {line}\n")
+                else:
+                    output.write(f"{line}\n")
+
+                if self._PATTERNS["done_line"].search(line):
+                    break
+        except StopIteration:
+            pass
+
+    def _process_refs(self, comment: str) -> str:
```

Comment:
Does not look like this is not used as well - I assume it is meant to flatten markdown URL's in comments.